### ISSUE TITLE
Resolved data type is wrong

### ISSUE THREAD

**Issue by peter-gribanov** [kalessil/phpinspectionsea#1510]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.3 |
| Language level | PHP 7.3                        |

The [`ceil()`](https://www.php.net/manual/en/function.ceil.php) function rounded up value to the next highest integer, but return a `float`. So i force convert a `float` value to `integer`, but *Php Inspections* does not see this.

```php
$round_time = (int) (ceil(time() / $cache_lifetime) * $cache_lifetime);
$cache_lifetime = $round_time - time();
```

Current behaviour
--

![image](https://user-images.githubusercontent.com/1954436/74433041-b351af00-4e70-11ea-83f9-b8da19c0f97e.png)

Expected behaviour
--

no error

Environment details
---

```
PhpStorm 2019.3.2
Build #PS-193.6015.48, built on January 22, 2020
```

**Comment by kalessil** [comment]
@peter-gribanov, thank you for reporting.

Would you please re-check the issue with PhpStorm 2019.3.3 (I can not reproduce it locally)?
Thank you in advance.

**Comment by peter-gribanov** [comment]
The problem is still relevant in PhpStorm 2019.3.3.

The problem is that the variable is already declared as integer. If the variable is not declared, then everything is fine.

Try this code:

```php
function roundCacheLifetime(int $cache_lifetime): int
{
    $round_time = (int) (ceil(time() / $cache_lifetime) * $cache_lifetime);
    $cache_lifetime = $round_time - time();
    $cache_lifetime = $cache_lifetime > 0 ? $cache_lifetime : 0;

    return $cache_lifetime;
}
```
![image](https://user-images.githubusercontent.com/1954436/75224738-1a992880-57ba-11ea-8e78-7a1a5de50779.png)

Declare another variable:

```php
function roundCacheLifetime(int $cache_lifetime): int
{
    $round_time = (int) (ceil(time() / $cache_lifetime) * $cache_lifetime);
    $round_cache_lifetime = $round_time - time();
    $round_cache_lifetime = $round_cache_lifetime > 0 ? $round_cache_lifetime : 0;

    return $round_cache_lifetime;
}
```

![image](https://user-images.githubusercontent.com/1954436/75224446-8dee6a80-57b9-11ea-8126-699366cbc874.png)

**Comment by kalessil** [comment]
@peter-gribanov: understood. Will re-evalaute.

**Comment by kalessil** [comment]
@peter-gribanov : I recall we fixed this in another ticket, but don't remember which one. So adding new test case to evaluate myself.
